app/src/test/java/org/mozilla/fenix/collections/CollectionCreationFragmentTest.kt

```diff
@@ -98,18 +93,41 @@ class CollectionCreationFragmentTest {
     }
 
     @Test
-    fun `GIVEN tabs are not present in session manager WHEN getTabs is called THEN an empty list will be returned`() {
-        val tabs = sessionManager
-            .getTabs(arrayOf(SESSION_ID_BAD_1, SESSION_ID_BAD_2), store, publicSuffixList)
+    fun `GIVEN tabs are not present in state WHEN getTabs is called THEN an empty list will be returned`() {
+        val tabs = state.getTabs(arrayOf(SESSION_ID_BAD_1, SESSION_ID_BAD_2), publicSuffixList)
 
         assertEquals(emptyList<Tab>(), tabs)
     }
 
     @Test
     fun `WHEN getTabs is called will null tabIds THEN an empty list will be returned`() {
-        val tabs = sessionManager
-            .getTabs(null, store, publicSuffixList)
+        val tabs = state.getTabs(null, publicSuffixList)
 
         assertEquals(emptyList<Tab>(), tabs)
     }
+
+    @Test
+    fun `toTab uses active reader URL`() {
```

Comment:
Hm this functionality should be independent of collections e.g. it should also be applied for tabs in the tabs tray, is ti?

Reply:
Actually, this works just as well, as before. The missing piece here is to update the URL for restored sessions, but didn't have that yet.